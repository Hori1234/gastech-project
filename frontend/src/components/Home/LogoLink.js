import React from "react";
import { Button, Image, Layout, Typography } from "antd";
import { Link } from "react-router-dom";
const { Text } = Typography;

/**
 * Displays a logo with a title and a description.
 *
 * Next to the logo is button which links to a component.
 *
 * @param image: Image to display with the link.
 * @param title: Title to display next to the image.
 * @param description: Text to display under the title.
 * @param to: Link to which the button should redirect the user to.
 */
export default function LogoLink({ image, title, description, to }) {
  return (
    <Layout
      style={{
        flexDirection: "row",
        alignItems: "center",
        display: "flex",
        marginBottom: 50,
        width: "100%",
        backgroundColor: "white",
      }}
    >
      <Image
        style={{ marginRight: 20 }}
        preview={false}
        width={100}
        src={image}
      />
      <Layout style={{ flexDirection: "column", backgroundColor: "white" }}>
        <Text style={{ fontWeight: "bold", fontSize: "18" }}>{title}</Text>
        <Text style={{ fontSize: " 14" }}>{description}</Text>
      </Layout>
      <Layout
        style={{
          alignItems: "flex-end",
          backgroundColor: "white",
          marginLeft: 20,
        }}
      >
        <Button id="dataButton" type="primary">
          <Link to={to}>Go</Link>
        </Button>
      </Layout>
    </Layout>
  );
}
