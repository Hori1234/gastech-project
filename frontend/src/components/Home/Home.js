import React, { Component } from "react";
import { AuthContext } from "../contextConfig.js";
import { Layout, Divider } from "antd";
import LogoLink from "./LogoLink";
import "antd/dist/antd.css";

/**
 * Displays the main menu component.
 *
 * The main menu displays a set of links with logo,
 * which redirect to other places in the application.
 *
 * Which links are displayed depends on the currently logged in user.
 */
export default class Home extends Component {
  render() {
    return (
      <AuthContext.Consumer>
        {(context) => (
          <Layout
            style={{
              flexDirection: "row",
              alignItems: "center",
              display: "flex",
              justifyContent: "space-around",
              backgroundColor: "white",
            }}
          >
            <Layout
              style={{
                alignItems: "flex-start",
                display: "flex",
                justifyContent: "space-around",
                marginBottom: 50,
                marginRight: 100,
                backgroundColor: "white",
                marginLeft: 10,
              }}
            >
              {(context.state.user.role === "planner" ||
                context.state.user.role === "administrator") && (
                <>
                  <LogoLink
                    image={require("../Images/Data Visualisation.webp")}
                    title={"Information panel"}
                    description={
                      "This view contains all the redable information related to this data challenge."
                    }
                    to={"/stats"}
                  />
                  <LogoLink
                    image={require("../Images/view-planning.svg")}
                    title={"Daily Overview"}
                    description={
                      "This page contains visualizaitons based on the daily connections and similarities between the employees."
                    }
                    to={"/expl"}
                  />
                  <LogoLink
                    image={require("../Images/Data Analytics.webp")}
                    title={"General Overview"}
                    description={
                      "This page contains visualizaitons based on the general connections between the employees."
                    }
                    to={"/hierarch"}
                  />
                </>
              )}
            </Layout>
            <Layout
              style={{
                height: "50vh",
                alignItems: "flex-end",
                backgroundColor: "white",
              }}
            >
              <Divider
                type="vertical"
                style={{ height: "50vh", marginRight: 20 }}
              />
            </Layout>
            <Layout
              style={{
                justifyContent: "space-around",
                marginBottom: 50,
                display: "flex",
                alignItems: "flex-start",
                backgroundColor: "white",
                marginRight: 20,
              }}
            >
              
              {context.state.user.role === "administrator" && (
                <LogoLink
                  image={require("../Images/1635626-200.png")}
                  title={"Account Management"}
                  description={
                    "Modify/add/delete of the accounts"
                  }
                  to={"/account"}
                />
              )}
            </Layout>
          </Layout>
        )}
      </AuthContext.Consumer>
    );
  }
}
