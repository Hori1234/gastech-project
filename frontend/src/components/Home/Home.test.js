import React from 'react';
import Home from './Home';
import { shallow } from 'enzyme';

describe('Home Component', () => {

    it('should render the login page correctly', () => {

        const component = shallow(<Home />);
        expect(component).toMatchSnapshot();
    });

});