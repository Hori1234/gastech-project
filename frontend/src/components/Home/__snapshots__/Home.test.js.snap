// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render the login page correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
